react-components/src/hooks/useClickedNode.tsx

```diff
@@ -54,58 +59,85 @@ export const useClickedNodeData = (): ClickedNodeData | undefined => {
+  ).data;
+
+  return useCombinedClickedNodeData(nodeDataPromises, assetMappingResult, cadIntersection);
+};
+
+const useCombinedClickedNodeData = (
+  fdmPromises: FdmNodeDataPromises | undefined,
+  assetMappings: NodeAssetMappingResult | undefined,
+  cadIntersection: CadIntersection | undefined
+): ClickedNodeData | undefined => {
+  const [clickedNodeData, setClickedNodeData] = useState<ClickedNodeData | undefined>();
+  const fdmData = useFdmData(fdmPromises);
+
   useEffect(() => {
-    void setClickedNodeFromQueryResult(nodeDataPromises);
+    if (cadIntersection === undefined) {
+      setClickedNodeData(undefined);
+      return;
+    }
 
-    async function setClickedNodeFromQueryResult(
-      promises: FdmNodeDataPromises | undefined
-    ): Promise<void> {
-      if (promises === undefined) {
-        return;
-      }
+    const assetMappingData =
+      assetMappings?.node === undefined
+        ? undefined
+        : {
+            cadNode: assetMappings.node,
+            assetIds: assetMappings.mappings.map((mapping) => mapping.assetId)
+          };
+
+    setClickedNodeData({
+      fdmResult: fdmData,
+      assetMappingResult: assetMappingData,
+      intersection: cadIntersection
+    });
+  }, [cadIntersection, fdmData, assetMappings]);
 
-      if (cadIntersection === undefined) {
-        setClickedNodeData(undefined);
-        return;
-      }
+  return clickedNodeData;
+};
 
-      setClickedNodeData({
-        intersection: cadIntersection
-      });
+const useFdmData = (
+  fdmPromises: FdmNodeDataPromises | undefined
+): FdmNodeDataResult | undefined => {
+  const [fdmData, setFdmData] = useState<FdmNodeDataResult | undefined>();
 
+  useEffect(() => {
+    if (fdmPromises === undefined) {
+      setFdmData(undefined);
+      return;
+    }
+
+    void setData(fdmPromises);
+
+    async function setData(promises: FdmNodeDataPromises): Promise<void> {
       const cadAndFdmNodes = await promises.cadAndFdmNodesPromise;
 
       if (cadAndFdmNodes === undefined || cadAndFdmNodes.fdmIds.length === 0) {
         return;
       }
 
-      setClickedNodeData({
-        intersection: cadIntersection,
+      setFdmData({
         fdmNode: cadAndFdmNodes.fdmIds[0],
         cadNode: cadAndFdmNodes.cadNode
       });
 
-      await setClickedNodeFromViewPromise(cadAndFdmNodes, promises.viewsPromise);
-    }
-
-    async function setClickedNodeFromViewPromise(
-      data: CadNodeWithFdmIds,
-      viewsPromise: Promise<Source[] | undefined>
-    ): Promise<void> {
-      const views = await viewsPromise;
+      const views = await promises.viewsPromise;
 
-      if (views === undefined || views.length === 0 || cadIntersection === undefined) {
+      if (views === undefined || views.length === 0) {
         return;
       }
 
-      setClickedNodeData({
-        intersection: cadIntersection,
-        fdmNode: data.fdmIds[0],
-        cadNode: data.cadNode,
+      setFdmData({
+        fdmNode: cadAndFdmNodes.fdmIds[0],
```

Comment:
Should we return all fdmIds here? The same applies to views.

Reply:
Agreed, since we're already changing the type signature of the function :+1: Good call